google-beta/resource_dataflow_flex_template_job_test.go

```diff
@@ -123,136 +151,87 @@ func testAccDataflowFlexTemplateJobGetGeneratedInstance(t *testing.T, s *terrafo
 	return instance, nil
 }
 
-// note: this config creates a job that doesn't actually do anything
-func testAccDataflowFlexTemplateJob_basic(bucket, job string) string {
+// note: this config creates a job that doesn't actually do anything, but still runs
+func testAccDataflowFlexTemplateJob_basic(job, topicName string) string {
 	return fmt.Sprintf(`
-resource "google_storage_bucket" "temp" {
-  name = "%s"
-  force_destroy = true
+data "google_storage_bucket_object" "flex_template" {
+  name   = "latest/flex/Streaming_Data_Generator"
+  bucket = "dataflow-templates"
 }
 
-resource "google_storage_bucket_object" "flex_template" {
-  name   = "flex_template.json"
-  bucket = google_storage_bucket.temp.name
-  content = <<EOF
-{
-    "image": "my-image",
-    "metadata": {
-        "description": "An Apache Beam streaming pipeline that reads JSON encoded messages from Pub/Sub, uses Beam SQL to transform the message data, and writes the results to a BigQuery",
-        "name": "Streaming Beam SQL",
-        "parameters": [
-            {
-                "helpText": "Pub/Sub subscription to read from.",
-                "label": "Pub/Sub input subscription.",
-                "name": "inputSubscription",
-                "regexes": [
-                    "[-_.a-zA-Z0-9]+"
-                ]
-            },
-            {
-                "helpText": "BigQuery table spec to write to, in the form 'project:dataset.table'.",
-                "is_optional": true,
-                "label": "BigQuery output table",
-                "name": "outputTable",
-                "regexes": [
-                    "[^:]+:[^.]+[.].+"
-                ]
-            }
-        ]
-    },
-    "sdkInfo": {
-        "language": "JAVA"
-    }
-}
-EOF
-}
-
-resource "google_dataflow_flex_template_job" "big_data" {
+resource "google_dataflow_flex_template_job" "job" {
   name = "%s"
-  container_spec_gcs_path = "${google_storage_bucket.temp.url}/${google_storage_bucket_object.flex_template.name}"
+  container_spec_gcs_path = "gs://${data.google_storage_bucket_object.flex_template.bucket}/${data.google_storage_bucket_object.flex_template.name}"
   on_delete = "cancel"
   parameters = {
-    inputSubscription = "my-subscription"
-    outputTable  = "my-project:my-dataset.my-table"
+    schemaLocation = "gs://mybucket/schema.json"
```

Comment:
Does this need to reference real objects and topics? This seems like it wouldn't successfully run

Reply:
Luckily, no, it doesn't need to reference real cloud resources. All the is required to update a job is that the existing job be in a RUNNING state. Using fake resource names for sources/sinks in the job causes the source and sink stages to emit errors while running, but the pipeline as a whole is still running.